main.py

```diff
@@ -26,6 +26,14 @@ def get_item(item_id: int):
     return data[item_id]
 
 
+@app.put("/update_item/{item_id}")
+def update_item(item_id: int, new_item: str):
+    if 0 <= item_id < len(data):
+        data[item_id] = new_item
+        return {"message": f"Item at index {item_id} updated successfully"}
+    else:
+        return {"message": "Item not found"}
```

Comment:
**Suggestion:** Check if the new_item is not empty or None before updating the data.

```suggestion
@app.put("/update_item/{item_id}")
def update_item(item_id: int, new_item: str):
    if new_item and 0 <= item_id < len(data):
        data[item_id] = new_item
        return {"message": f"Item at index {item_id} updated successfully"}
    else:
        return {"message": "Item not found"}
```